tsa-test/src/test/kotlin/org/ton/examples/intercontract/SendModesTest.kt

```diff
@@ -371,19 +388,26 @@ class SendModesTest {
             TvmOptions(
                 intercontractOptions = IntercontractOptions(communicationScheme = communicationScheme),
                 enableOutMessageAnalysis = true,
+                stopOnFirstError = false,
             )
 
         val tests =
             analyzeInterContract(
                 listOf(checkerContract, analyzedSender, analyzedRecipient),
+                concreteContractData =
+                    listOf(
+                        TvmConcreteContractData(contractC4 = CellBuilder().storeInt(opcode, 64).endCell()),
+                        TvmConcreteContractData(),
+                        TvmConcreteContractData(),
+                    ),
                 startContractId = 0,
                 methodId = TvmContext.RECEIVE_INTERNAL_ID,
                 options = options,
             )
 
         propertiesFound(
             tests,
-            listOf { test -> test.exitCode() == 258 },
+            listOf { test -> test.exitCode() == 500 },
         )
```

Comment:
We probably also need checks here for `events` (maybe in `checkInvariants`). Seems like currently these checks are too weak.

Reply:
I've hardened them somewhat --- now I also check that the message was/wasn't sent depending on the base test parameter. 
Please see if this is enough.